src/warehouses/components/WarehouseSettings/WarehouseSettings.tsx

```diff
@@ -0,0 +1,200 @@
+    {
+      label: (
+        <>
+          <FormattedMessage defaultMessage="Disabled" />
+          <Typography variant="caption" color="textSecondary">
+            <FormattedMessage defaultMessage="If selected customer won't be able to choose this warehouse as pickup point" />
+          </Typography>
+          <FormSpacer />
+        </>
+      ),
+      value: WarehouseClickAndCollectOptionEnum.DISABLED
+    },
+    {
+      label: (
+        <>
+          <FormattedMessage defaultMessage="Local stock only" />
+          <Typography variant="caption" color="textSecondary">
+            <FormattedMessage defaultMessage="If selected customer will be able to choose this warehouse as pickup point. Ordered products will be only fulfilled from this warehouse stock" />
+          </Typography>
+          <FormSpacer />
+        </>
+      ),
+      value: WarehouseClickAndCollectOptionEnum.LOCAL
+    },
+    {
+      label: (
+        <>
+          <FormattedMessage defaultMessage="All warehouses" />
+          <Typography variant="caption" color="textSecondary">
+            <FormattedMessage defaultMessage="If selected customer will be able to choose this warehouse as pickup point. Ordered products can be shipped here from a different warehouse" />
+          </Typography>
+        </>
+      ),
+      value: WarehouseClickAndCollectOptionEnum.ALL
+    }
+  ];
+
+  const clickAndCollectChoicesPrivate = clickAndCollectChoicesPublic.filter(
+    choice => choice.value !== WarehouseClickAndCollectOptionEnum.LOCAL
+  );
+
+  return (
+    <Card>
+      <CardTitle
+        title={intl.formatMessage({
+          defaultMessage: "Settings",
+          description: "zones that warehouse sends to"
+        })}
+      />
+      <CardContent>
+        {renderCollection(
+          zones,
+          zone =>
+            zone ? (
+              <div className={classes.link} key={zone.id}>
+                <Link underline onClick={() => onShippingZoneClick(zone.id)}>
+                  {zone.name}
+                </Link>
+              </div>
+            ) : (
+              <Skeleton />
+            ),
+          () => (
+            <Typography color="textSecondary">
+              <FormattedMessage defaultMessage="This warehouse has no shipping zones assigned." />
+            </Typography>
+          )
+        )}
+      </CardContent>
+      <Divider />
+      <CardContent>
+        <CardSpacer />
+        <RadioGroupField
+          disabled={disabled}
+          choices={isPrivateChoices}
+          onChange={onChange}
+          value={data.isPrivate}
+          name="isPrivate"
+          alignTop={true}
+        />
+      </CardContent>
+      <Divider />
+      <CardContent>
+        <Typography color="textSecondary" variant="h6">
+          <FormattedMessage defaultMessage="Pickup" />
+        </Typography>
+        <CardSpacer />
+        {data.isPrivate === "true" ? (
+          <RadioGroupField
+            disabled={disabled}
```

Comment:
actually all the props in this radio group field besides choices are the same, why not do
```suggestion
  <RadioGroupField
            disabled={disabled}
            onChange={onChange}
            value={data.clickAndCollectOption}
            name="clickAndCollectOption"
            alignTop={true}
            choices={data.isPrivate === "true" ?
               clickAndCollectChoicesPrivate : clickAndCollectChoicesPublic
            }
          />
```

Reply:
Done.